tutorials/nginx-ingress-gke/index.md

```diff
@@ -0,0 +1,353 @@
+
+	 annotations: kubernetes.io/ingress.class: nginx
+
+
+On Kubernetes Engine, if no annotation is defined under the metadata, the
+Ingress Resource uses the GCP GCLB L7 load balancer to serve traffic.  This
+method can also be forced by setting  the annotation's value to `gce`.
+
+	annotations: kubernetes.io/ingress.class: gce
+
+You can verify the annotation by viewing the ingress-resource.yaml file and check the annotations under the metadata section as shown below.
+
+	cat ingress-resource.yaml
+
+You will see:
+
+	apiVersion: extensions/v1beta1
+	kind: Ingress
+	metadata:
+	  name: ingress-resource
+	  annotations:
+	    kubernetes.io/ingress.class: nginx
+	spec:
+	  rules:
+	  - http:
+	      paths:
+	      - path: /
+		backend:
+		  serviceName: kuard
+		  servicePort: 80
+
+The kind: Ingress dictates it is an Ingress Resource object.  This Ingress Resource defines an inbound L7 rule for path / to service kuard on port 80.
+
+From the Cloud Shell.  
+  
+	kubectl apply -f ingress-resource.yaml
+
+Verify the Ingress Resource has been created.  Please note that the IP address
+for the Ingress Resource has not yet been defined.  
+  
+	kubectl get ingress ingress-resource  
+  
+	NAME               HOSTS     ADDRESS   PORTS     AGE  
+	ingress-resource   *                   80        `
+
+Now that we have an Ingress Resource defined, we need an Ingress Controller to
+act upon the rules as shown below.
+
+<img src="https://github.com/ameer00/community/blob/master/tutorials/nginx-ingress-gke/Nginx%20Ingress%20on%20GCP%20-%20Fig%2001.png" width="65%">
+
+# Deploying the nginx Ingress Controller
+
+Kubernetes platform allows for administrators to bring their own Ingress
+Controllers instead of using the cloud provider's built-in offering. 
+
+The Nginx controller, deployed as a service, must be exposed for external
+access.  This is done using service _Type: LoadBalancer_ on the nginx controller
+service.  On GKE, this creates a Google Cloud L4 (TCP) load balancer with nginx
+controller service as a backend.  Google Cloud also creates the appropriate
+firewall rules within the service's VPC to allow web HTTP(S) traffic to the load
+balancer frontend IP address.  Here is a basic flow of the nginx ingress
+solution on GKE.
+
+## Nginx Ingress Controller on GKE
+
+<img src="https://github.com/ameer00/community/blob/master/tutorials/nginx-ingress-gke/Nginx%20Ingress%20on%20GCP%20-%20Fig%2002.png" width="65%">
+
+From the Cloud Shell, let us deploy an Nginx controller and service.  
```

Comment:
Change all periods before command samples, code samples, and diagrams, to colons throughout.